.github/workflows/builder.yaml

```diff
@@ -0,0 +1,54 @@
+name: CI Build & Push Builder Image
+on:
+  pull_request_target:
+    types: [ opened, synchronize, reopened ]
+    paths:
+    - Dockerfile.builder
+    - .bazelversion
+
+jobs:
+  build-and-push-builder:
+    timeout-minutes: 30
+    name: Build and push multi-arch builder images
+    runs-on: ubuntu-latest
+    environment: release-base-images
+    steps:
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@8c0edbc76e98fa90f69d9a2c020dcb50019dc325
+      with:
+        version: v0.9.1
+    - name: Cache Docker layers
+      uses: actions/cache@58c146cc91c5b9e778e71775dfe9bf1442ad9a12
+      with:
+        path: /tmp/buildx-cache
+        key: docker-cache-${{ github.head_ref }}
+        restore-keys: docker-cache-builder-master
+    - name: Login to quay.io
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+      with:
+        registry: quay.io
+        username: ${{ secrets.QUAY_ENVOY_USERNAME }}
+        password: ${{ secrets.QUAY_ENVOY_PASSWORD }}
+    - name: Checkout PR
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Prep for build
+      run: |
+        echo "${{ github.event.pull_request.head.sha }}" >SOURCE_VERSION
```

Comment:
The `SOURCE_VERSION` should be built from `git ls-tree --full-tree HEAD -- DIRECTORY | awk '{ print $3 }'` this `DIRECTORY` is a directory that should contain `Dockerfile.builder` and `.bazelversion` which are the files used to derive the SHA. This SHA will always be the same as long the files are NOT modified.